src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -110,6 +113,72 @@ object HttpGoogleServicesDAO extends GoogleServicesDAO with FireCloudRequestBuil
     googleCredential.getAccessToken
   }
 
+  override def getAdminIdentityToken(implicit actorRefFactory: ActorRefFactory): Future[String] = {
+
+    // https://cloud.google.com/iap/docs/authentication-howto#authenticating_from_a_service_account
+
+    val OAUTH_TOKEN_URI = "https://www.googleapis.com/oauth2/v4/token"
+
+    // get private key, private key id, and email for the orch service account
+    val googleCredential = new GoogleCredential.Builder()
+      .setTransport(httpTransport)
+      .setJsonFactory(jsonFactory)
+      .setServiceAccountId(pemFileClientId)
+      .setServiceAccountScopes(Seq("openid", "email")) // use the smallest scope possible
+      .setServiceAccountPrivateKeyFromPemFile(new java.io.File(pemFile))
+      .build()
+    val privateKeyId = googleCredential.getServiceAccountPrivateKeyId
+    val privateKey = googleCredential.getServiceAccountPrivateKey
+    val email = googleCredential.getServiceAccountId
+
+    // 1. self-sign a service account JWT with the target_audience claim set to the URL of the receiving function.
+
+    // build JWT header
+    val header = JwtHeader(Option(JwtAlgorithm.RS256), Option("JWT"), None, Option(privateKeyId))
+
+    // build JWT claim
+    val iat:Long = Instant.now().getEpochSecond()
+    val exp:Long = iat + 3600
+    val claim = JwtClaim(
+      content = s"""{"target_audience": "${EntityClient.avroToRawlsURL}"}""",
+      issuer = Option(email),
+      subject = Option(email),
+      audience = Option(Set(OAUTH_TOKEN_URI)),
+      expiration = Option(exp),
+      issuedAt = Option(iat)
+    )
+
+    // build JWT itself
+    val jwt = Jwt.encode(header, claim, privateKey)
+
+    // 2. Exchange the self-signed JWT for a Google-signed ID token, which should have the aud claim set to the above URL.
+    val tokenPayload =
+      s"""
+        | {
+        |   "grant_type": "urn:ietf:params:oauth:grant-type:jwt-bearer",
+        |   "assertion": "${jwt.toString}",
+        |   "scope": "openid email"
+        | }
+        |""".stripMargin
+
+    val oidcPipeline = sendReceive
+    val oidcRequest = Post(OAUTH_TOKEN_URI, tokenPayload)
+
+    oidcPipeline{ oidcRequest } map { resp: HttpResponse =>
+      resp.status match {
+        case OK =>
+          val oidcJson = resp.entity.asString.parseJson.asJsObject
+          oidcJson.fields.get("id_token") match {
+            case Some(s:JsString) => s.value
+            case _ => throw new FireCloudException(s"id_token string value not found in OIDC token: ${oidcJson.fields.keySet}")
+          }
+        case badStatus =>
+          logger.error(resp.message.toString)
+          throw new FireCloudException(s"Error $badStatus getting OIDC token")
+      }
+    }
+  }
+
```

Comment:
I'm surprised that I can't find this in the Google client libraries. Since all of Google's docs around identity tokens (see PR description) also have examples that manually build JWTs and call the endpoint, I went with that approach too.